src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -196,26 +196,13 @@ Vector MotionController::determineLinearVelocityFromVelocity(
     const double max_acceleration_meters_per_second_squared, const double delta_time,
     const double max_speed_meters_per_second)
 {
-    double velocity_to_add = max_acceleration_meters_per_second_squared * delta_time;
     double new_velocity_x, new_velocity_y;
 
-    if (robot.velocity().x() < linear_velocity.x())
-    {
-        new_velocity_x = robot.velocity().x() + velocity_to_add;
-    }
-    else
-    {
-        new_velocity_x = robot.velocity().x() - velocity_to_add;
-    }
-
-    if (robot.velocity().y() < linear_velocity.y())
-    {
-        new_velocity_y = robot.velocity().y() + velocity_to_add;
-    }
-    else
-    {
-        new_velocity_y = robot.velocity().y() - velocity_to_add;
-    }
+    // the following method works best for determining how much to adjust the velocity
+    // https://courses.cs.washington.edu/courses/cse466/11au/calendar/10-control-posted.pdf
+    // refer to slide 11
+    new_velocity_x = -(robot.velocity().x() - linear_velocity.x());
+    new_velocity_y = -(robot.velocity().y() - linear_velocity.y());
```

Comment:
Looking at this again, I am a little surprised this works, just because this formula is returning the change in velocity, but we're assigning it to the velocity directly? If it works then that's fine, might just indicate there are some other issues with the motion controller.

Reply:
when I apply a unit vector, it perfectly holds at 1m/s speed on grsim (until it bumps into the wall in which case it corrects and then again holds at 1m/s). I was thinking maybe the reason its working is once you get to the initial 1m/s, u can apply a velocity vector with 0 magnitude as long as you are coasting at the right speed. So its only assigning the change it needs to get back onto the target velocity. 

But again, im not really sure about how the motion controller should work so maybe I should just open a ticket for someone else to look at it, and just finish pivot with these changes, as no other primitive uses the velocity based planner.

Reply:
Honestly, if the motion is smoother let's go with it. Especially since this is the only primitive that depends on it :stuck_out_tongue: